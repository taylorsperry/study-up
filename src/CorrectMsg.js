import React from "react"

const CorrectMsg = props => {
    return (
       <section className="quiz-card">
        <h2 className="correct">That's right!</h2>
        <p className="explanation">{props.explanation}</p>
        <button className="keep-practicing" onClick={props.keepPracticing}>Keep Practicing</button>
       </section> 
    )
}
export default CorrectMsg