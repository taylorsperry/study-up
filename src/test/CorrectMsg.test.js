import React from "react";
import CorrectMsg from "../CorrectMsg";
import { shallow } from "enzyme"

describe("CorrectMsg", () => {
  let wrapper;
  let currentQuiz = {
    "id": 3,
    "question": "If the array 'greeting' has a value of ['Hello', 'World'], and you wanted to return the string 'Hello World', which of the following would you call?",
    "answers": ["greeting.join()", "greeting.concat()", "greeting.join(' ')"],
    "correctAnswer": "greeting.join(' ')",
    "explanation": "Without a separator argument, .join() will return a string with the arrayâ€™s elements separated by commas; .concat() joins the prototype on which its called with the arguments its passed."
  }

  beforeEach(() => {
    wrapper = shallow(
    <CorrectMsg 
        explanation={currentQuiz.explanation}
    />
    );
  })

  it("should match snapshot when all data is passed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
})