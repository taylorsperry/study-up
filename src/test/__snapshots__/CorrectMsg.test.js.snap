// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CorrectMsg should match snapshot when all data is passed correctly 1`] = `
<section
  className="quiz-card"
>
  <h2
    className="correct"
  >
    That's right!
  </h2>
  <p
    className="explanation"
  >
    Without a separator argument, .join() will return a string with the arrayâ€™s elements separated by commas; .concat() joins the prototype on which its called with the arguments its passed.
  </p>
  <button
    className="keep-practicing"
  >
    Keep Practicing
  </button>
</section>
`;
